[Scripts] Toybox Cleanup & Organization

imgntn opened the pull request:
This PR:
-renames /toys to /toybox,
-relocates non toybox scripts,
-updates master refs, 
-updates script refs

Test by loading this script in toybox and then clicking the arcade machine to reset.
https://rawgit.com/imgntn/hifi/cleanup_toybox/unpublishedScripts/immediateClientReset.js

ZappoMan commented:
@imgntn - this PR moves around a bunch of files into places they really shouldn't be... why did you do this? Who asked for this? In particular I am concerned that you've moved a bunch of files into "entityScripts" directories when the files themselves are not entity scripts... Can you repair this? Let's not put files that are not actual entity scripts into directories with that name. examples: magBalls, breakdanceToy.js, grenade.js,

imgntn commented:
@ZappoMan you're right, except maybe the breakdancer entity, those don't really belong in entityScripts.  I moved magBalls, breakdancing (except the entity), and grenade to the top level, and adjusted the paths in this PR https://github.com/highfidelity/hifi/pull/6163
